driver-core/src/test/java/com/datastax/driver/core/AuthenticationTest.java

```diff
@@ -132,4 +137,29 @@ public void run() {
         }, 2000);
     }
 
+    /**
+     * Ensures that the driver throws an AuthenticationException
+     * when an authentication error occurs during connection pool initialization.
+     *
+     * @jira_ticket JAVA-1431
+     */
+    @Test(groups = "short")
+    @CCMConfig(dirtiesContext = true)
+    public void should_not_connect_with_wrong_credentials() throws InterruptedException {
+        PlainTextAuthProvider authProvider = new PlainTextAuthProvider("cassandra", "cassandra");
+        Cluster cluster = Cluster.builder()
+            .addContactPoints(getContactPoints())
+            .withPort(ccm().getBinaryPort())
+            .withAuthProvider(authProvider)
+            .build();
+        cluster.init();
+        authProvider.setPassword("wrong");
```

Comment:
> I think triggering the host down and starting reconnection and not propagating the exception up to the user might be a good happy medium as you can then establish connections to the rest of the nodes (in this case). wdyt?

`AuthenticationException` is a strange kind of error, because it can be specific to a host in some cases (like in your example), or cluster-wise in some others; besides, it can be sometimes temporary (like when a user changes his password), sometimes permanent (wrong credentials).

But you might be right, I tested your suggestion and it seems a good comprise. On one side, it does not prevent the session from being created, which is good in your scenario. But on the other side, it also improves the user experience by having the session throw NHAE instead of `BusyPoolException` if the error happened cluster-wise.

So yes, I think you have a very good point, I will amend the code accordingly, thanks!